app/controllers/api/mixins/operations.rb

```diff
@@ -0,0 +1,65 @@
+module Api
+  module Mixins
+    #
+    # Contains the routines to enqueue operations for resources
+    #
+    module Operations
+      def perform_action(action, type, id)
+        if single_resource?
+          enqueue_action_single_resource(action, type, id)
+        else
+          enqueue_action_multiple_resources(action, type, id)
+        end
+      end
+
+      #
+      # Enqueues the action for a single resource.
+      #
+      # @param [symbol] action - action to be enqueued
+      # @param [symbol] type   - type of the resource
+      # @param [number] id     - id of the resource
+      #
+      def enqueue_action_single_resource(action, type, id)
+        raise BadRequestError, "Must specify an id for changing a #{type} resource" unless id
+
+        resource = resource_search(id, type, collection_class(type))
+
+        api_action(type, id) do
+          begin
+            desc = "Performing #{action} for #{resource_identify(resource)}"
+            api_log_info(desc)
+            task_id = queue_object_action(resource, desc, :method_name => action, :role => :ems_operations)
+            action_result(true, desc, :task_id => task_id)
+          rescue StandardError => err
+            action_result(false, err.to_s)
+          end
+        end
+      end
+
+      #
+      # Enqueues the action for multiple resources.
+      #   For multiple resources, when an error occurs, the error messages must
+      #   be built individually for each resource. Always responding with status 200.
+      #
+      # @param [symbol] action - action to be enqueued
+      # @param [symbol] type   - type of the resource
+      # @param [number] id     - id of the resource
+      #
+      def enqueue_action_multiple_resources(action, type, id)
+        enqueue_action_single_resource(action, type, id)
+      rescue ActiveRecord::RecordNotFound => err
+        action_result(false, _(err.message))
```

Comment:
This doesn't seem to match up with the comment, since you aren't building any error message (and for that matter only have 1 resource).  I realize this is a copy-paste from the physical_switches_controller, but if it's being promoted to a common mixin, it needs to be consistent with other modules in the base.

Reply:
Actually, even the request is made for multiple resources, this method is called individually for each one. With `built individually`, in the comment, I mean to call `action_result(false, _(err.message))` each time that occurs an error for some resource, which is what the method does. It is a copy-paste from physical_switches_controller, but with some adaptations to work with other resources, we can see this on this PR https://github.com/ManageIQ/manageiq-api/pull/410 where physical_chassis_controller also uses this routine.